src/services/getEditsForFileRename.ts

```diff
@@ -1,67 +1,159 @@
+     * An import that comes from inside the directory and resolves to inside the directory won't need to be updated.
+     */
+    type InternalPathUpdater = (sourceFile: SourceFile, importText: string) => string | undefined;
+    function getInternalPathUpdater(oldFileOrDirPath: string, newFileOrDirPath: string): InternalPathUpdater {
+        const relativeNewDirToOldDir = getRelative(newFileOrDirPath, oldFileOrDirPath);
+        if (relativeNewDirToOldDir === ".") return () => undefined;
+        const relativeOldDirToNewDir = getRelative(oldFileOrDirPath, newFileOrDirPath);
 
-            for (const importStringLiteral of sourceFile.imports) {
-                const resolved = host.resolveModuleNames
-                    ? host.getResolvedModuleWithFailedLookupLocationsFromCache && host.getResolvedModuleWithFailedLookupLocationsFromCache(importStringLiteral.text, sourceFile.fileName)
-                    : program.getResolvedModuleWithFailedLookupLocationsFromCache(importStringLiteral.text, sourceFile.fileName);
-                // We may or may not have picked up on the file being renamed, so maybe successfully resolved to oldFilePath, or maybe that's in failedLookupLocations
-                if (resolved && contains(resolved.resolvedModule ? [resolved.resolvedModule.resolvedFileName] : resolved.failedLookupLocations, oldFilePath)) {
-                    result.push({ sourceFile, toUpdate: importStringLiteral });
-                }
+        return (sourceFile, importText) => {
+            if (!pathIsRelative(importText) ||
+                importIsInternalToDirectory(oldFileOrDirPath, sourceFile.fileName, importText) ||
+                importIsInternalToDirectory(newFileOrDirPath, sourceFile.fileName, importText)) {
+                return undefined;
             }
-        }
-        return result;
+            // If we're renaming "./a" to "./foo/a", an import from "./foo/z" can just import "./a".
+            const short = tryRemovePrefix(importText, relativeOldDirToNewDir + "/");
+            if (short !== undefined) return ensurePathIsNonModuleName(short);
+            return combineNormal(relativeNewDirToOldDir, importText);
+        };
     }
 
-    function getPathUpdater(oldFilePath: string, newFilePath: string, host: LanguageServiceHost): (oldPath: string) => string | undefined {
+    function getRelative(from: string, to: string): string {
+        return ensurePathIsNonModuleName(getRelativePathFromDirectory(toDirectory(from), toDirectory(to), /*ignoreCase*/ false));
+    }
+
+    function toDirectory(path: string): string {
+        return isAnySupportedFileExtension(path) ? getDirectoryPath(path) : path;
+    }
+
+    function importIsInternalToDirectory(oldDir: string, sourceFilePath: string, importText: string): boolean {
+        return !!removeDirectoryPrefixFromPath(oldDir, combineNormal(getDirectoryPath(sourceFilePath), importText));

```

Comment:
For all these operations you want to take into account the case sensitivity as well (since you are using fileName instead of path)